### ISSUE TITLE
Monospace check 079 is incorrectly triggered by propotional font in v 0.6.11

### ISSUE THREAD

**Issue by thundernixon** [googlefonts/fontbakery#2366]
## Observed behaviour

Currently, in Fontbakery 0.6.11, check 79 tells me that Encode Sans seems to be a monospace, with a self-contradicting message:

```
⚠️ WARN: Monospace font has hhea.advanceWidthMax equal to each glyph's advanceWidth?
⚠️ WARN This seems to be a monospaced font, so advanceWidth value should be the same across all glyphs, but 99.67% of them have a different value
```

![image](https://user-images.githubusercontent.com/7355414/53205265-ce8e3380-35fb-11e9-8463-c4328e988633.png)

This happens with both Condensed and Expanded versions of Encode.

I suspect that we may have set the math slightly wrong in the monospace check. If 99.67% of glyphs have a different width value, a font is almost certainly not intended to be monospace.

## Expected behaviour

**Guestimates**

I expect that a font with anything above maybe 20% or so of glyphs with variable width is not at all trying to be monospace. A font is probably monospace if 95%+ of glyphs have the same width.

If the check were even smarter, it could also detect the amount of variation in metrics. That is, if a bunch of glyphs are within a small range of widths (say, ±5% or so), it may be a sloppy monospace font.

**One example of real metrics that should fail**

Roboto Mono, which is on the more-variable side of monospace fonts, has about 3.25% of its glyphs with a width other than 1229 units (out 2048 UPM) in the upright Medium instance. In the Medium Italic instance, there are also 3.25% of its glyphs with widths different from the rest of the glyphs, _but_ the main width here is 1202, which is different from the upright glyphs.

See https://github.com/google/fonts/issues/1832 for more details.

Taken together, these fonts should certainly fail this check. Matching widths between styles is one thing (and I've filed a separate issue specifically for this at https://github.com/googlefonts/fontbakery/issues/2365). However, even these internal consistencies should fail check 079.

However, Encode Sans is very much a proportional design, and should not trigger this warning.

## Resources and exact process needed to replicate

I am specifically seeing this warning while I build split-width variable fonts in https://github.com/thundernixon/Encode-Sans.

It was not the case in Fontbakery `0.6.10` ([example checks](https://github.com/thundernixon/Encode-Sans/blob/5d034c6ba4c3764764487eb201f73375e425d494/fonts/encodesanscondensed/split_vf/EncodeSansCondensed-Thin-fontbakery-report.md)), but it is now an issue with v `0.6.11` ([example checks](https://github.com/thundernixon/Encode-Sans/blob/f1850a1fa58c07efc623fa96ba09bbd6428a74d1/fonts/encodesanscondensed/split_vf/EncodeSansCondensed-Thin-fontbakery-report.md)).

**Comment by m4rc1e** [comment]
In #2349 I refactored check 033 so only the ascii glyphs are used to determine whether a font is monospace. I did this so CJK fonts which contain proportional Latin glyphs are not marked as monospace.

I will need to update this check as well.

**Comment by m4rc1e** [comment]
Interestingly, the shared condition `glyph_metrics_stats` is never used in the [check]( https://github.com/googlefonts/fontbakery/blob/master/Lib/fontbakery/specifications/hhea.py#L48). this feels like a pointless check but I'll see what I can do next week.

**Comment by thundernixon** [comment]
To me, this (or something like it) seems like a really important check. It causes a bunch of little messes when a "monospace" font like Roboto Mono is released without being checked for full mono-spacedness.

That makes sense to need a way to avoid flagging CJK fonts as monospace, and shows that we can't rely on a percentage of all glyphs. Limiting check to just ASCII characters seems a bit limited, though, and would (I think) miss most of the issues in Roboto Mono. Maybe we could do something like checking the whole latin-ext set, or something that would cast a bit of a bigger net, but still avoid checking CJK glyphs?

**Comment by m4rc1e** [comment]
Fixed in #2368.
